### ISSUE TITLE
Can't load OpenAPI schema with DRF 3.14 and drf-spectacular 0.24.2, python 3.915

### ISSUE THREAD

**Issue by hamishfagg** [ghazi-git/drf-standardized-errors#20]
With the above package versions I get the following error when I try to load /schema/

Any idea what's going on here? 

```
172.24.0.1 - - [08/Dec/2022 15:48:00] "GET /schema/ HTTP/1.1" 500 -
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/django/core/handlers/wsgi.py", line 132, in __call__
    response = self.get_response(request)
  File "/usr/local/lib/python3.9/site-packages/django/core/handlers/base.py", line 140, in get_response
    response = self._middleware_chain(request)
  File "/usr/local/lib/python3.9/site-packages/django/core/handlers/exception.py", line 57, in inner
    response = response_for_exception(request, exc)
  File "/usr/local/lib/python3.9/site-packages/django/core/handlers/exception.py", line 139, in response_for_exception
    response = handle_uncaught_exception(
  File "/usr/local/lib/python3.9/site-packages/django/core/handlers/exception.py", line 180, in handle_uncaught_exception
    return debug.technical_500_response(request, *exc_info)
  File "/usr/local/lib/python3.9/site-packages/django_extensions/management/technical_response.py", line 40, in null_technical_500_response
    raise exc_value.with_traceback(tb)
  File "/usr/local/lib/python3.9/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
  File "/usr/local/lib/python3.9/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/local/lib/python3.9/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
    return view_func(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/django/views/generic/base.py", line 84, in view
    return self.dispatch(request, *args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/rest_framework/views.py", line 509, in dispatch
    response = self.handle_exception(exc)
  File "/usr/local/lib/python3.9/site-packages/rest_framework/views.py", line 469, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/usr/local/lib/python3.9/site-packages/rest_framework/views.py", line 480, in raise_uncaught_exception
    raise exc
  File "/usr/local/lib/python3.9/site-packages/rest_framework/views.py", line 506, in dispatch
    response = handler(request, *args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/drf_spectacular/views.py", line 82, in get
    return self._get_schema_response(request)
  File "/usr/local/lib/python3.9/site-packages/drf_spectacular/views.py", line 90, in _get_schema_response
    data=generator.get_schema(request=request, public=self.serve_public),
  File "/usr/local/lib/python3.9/site-packages/drf_spectacular/generators.py", line 268, in get_schema
    paths=self.parse(request, public),
  File "/usr/local/lib/python3.9/site-packages/drf_spectacular/generators.py", line 239, in parse
    operation = view.schema.get_operation(
  File "/usr/local/lib/python3.9/site-packages/drf_spectacular/utils.py", line 364, in get_operation
    return super().get_operation(path, path_regex, path_prefix, method, registry)
  File "/usr/local/lib/python3.9/site-packages/drf_spectacular/openapi.py", line 94, in get_operation
    operation['responses'] = self._get_response_bodies()
  File "/usr/local/lib/python3.9/site-packages/drf_standardized_errors/openapi.py", line 55, in _get_response_bodies
    serializer = self._get_error_response_serializer(status_code)
  File "/usr/local/lib/python3.9/site-packages/drf_standardized_errors/openapi.py", line 238, in _get_error_response_serializer
    return self._get_http400_serializer()
  File "/usr/local/lib/python3.9/site-packages/drf_standardized_errors/openapi.py", line 260, in _get_http400_serializer
    serializer = self._get_serializer_for_validation_error_response()
  File "/usr/local/lib/python3.9/site-packages/drf_standardized_errors/openapi.py", line 275, in _get_serializer_for_validation_error_response
    fields_with_error_codes = self._determine_fields_with_error_codes()
  File "/usr/local/lib/python3.9/site-packages/drf_standardized_errors/openapi.py", line 283, in _determine_fields_with_error_codes
    fields = get_flat_serializer_fields(serializer)
  File "/usr/local/lib/python3.9/site-packages/drf_standardized_errors/openapi_utils.py", line 77, in get_flat_serializer_fields
    name = get_prefix(prefix, field.field_name)
AttributeError: 'OpenApiTypes' object has no attribute 'field_name'
```

**Comment by ghazi-git** [comment]
Thank you for reporting the issue @hamishfagg.

I was able to reproduce it by adding `@extend_schema(request=OpenApiTypes.STR)` to an update view. So, can you please confirm that similar code `@extend_schema(request=OpenApiTypes....)` is also causing the error for you?

**Comment by ghazi-git** [comment]
Just released v0.12.4 which should fix this issue. If that's not the case, feel free to reopen the issue.

**Comment by hamishfagg** [comment]
Thanks very much for this.

For future googlers I ran into another error after installing 0.12.4:

`AssertionError: May not set both 'read_only' and 'write_only'`

This was due to having model fields listed as `write_only`, as well as listed in a `write_once` mixin I'm using. Removing one of these listings for the fields solved this issue.
